src/streamlink/plugins/ard_mediathek.py

```diff
@@ -100,24 +105,27 @@ def _get_streams(self):
                         continue
                     stream_ = stream_[0]
 
+                stream_ = update_scheme("https://", stream_)
                 if stream_.endswith(".f4m"):
                     parser = self._get_hds_streams
                     parser_name = "HDS"
                 elif stream_.endswith(".smil"):
                     parser = self._get_smil_streams
                     parser_name = "SMIL"
-                elif stream_.endswith(".m3u8"):
+                elif ".m3u8" in stream_:
                     parser = self._get_hls_streams
                     parser_name = "HLS"
                 elif stream_.startswith("http"):
                     parser = self._get_http_streams
                     parser_name = "HTTP"
+                else:
+                    raise PluginError("Unexpected stream type: '{0}'".format(stream_))
 
                 try:
                     for s in parser(stream):
                         yield s
                 except IOError as err:
-                    self.logger.error("Failed to extract {0} streams: {1}",
+                    raise PluginError("Failed to extract {0} streams: {1}",
```

Comment:
same here

Reply:
If there is only a log message, you might not recognize the error.